### PR TITLE
test(delivery-node, delivery-x-domain-request): convert tests

### PR DESCRIPTION
**Pull request by djskinner**

### CODE REVIEW

File: babel.config.js [hunk 1]
```diff
@@ -8,7 +8,8 @@ module.exports = api => {
   if (api && api.env('test')) {
     presets.push('@babel/preset-typescript')
     plugins.push(['@babel/plugin-proposal-class-properties', { loose: true }])
-    plugins.push(['@babel/plugin-transform-modules-commonjs'])
+    plugins.push('@babel/plugin-transform-modules-commonjs')
+    plugins.push('@babel/plugin-proposal-optional-chaining')
```
Comment (djskinner) [inline babel.config.js:12]: allows optional chaining in tests e.g. `expect(err?.meesage).toBe(` which is what the lint rules recommend

File: packages/delivery-node/delivery.d.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import { Delivery } from '@bugsnag/core/client'
```
Comment (djskinner) [inline packages/delivery-node/delivery.d.ts:1]: This works but I don't think is ideal. It's easy enough to use the internal types within the core package but externall we either need to export them or do this. I'm inclined to prefer exporting the `Delivery` type publically but this works for now.

File: packages/delivery-node/test/delivery.test.ts [hunk 3]
```diff
@@ -1,10 +1,18 @@
-const { describe, it, expect } = global
+import delivery from '../'
+import http from 'http'
+import { Client } from '@bugsnag/core'
+import { EventDeliveryPayload } from '@bugsnag/core/client'
```
Comment (djskinner) [inline packages/delivery-node/test/delivery.test.ts:4]: Same here RE internal types

**Review by twometresteve — COMMENTED** [review]
Happy that tests have been considered here (it's the whole point!).

File: packages/delivery-node/delivery.d.ts (hunk 2 above)
Comment (bengourley) [reply packages/delivery-node/delivery.d.ts:1]: discussed elsewhere and agreed this is fine

**Review by bengourley — APPROVED** [review]

**Pull request merged**